inputs/org.osbuild.containers-storage

```diff
@@ -0,0 +1,145 @@
+and a container ID for the stage to use.
+
+Unlike all the other input types, this does not refer to static
+objects in the osbuild sources store but a dynamically mounted
+container store that depends on the availability of the
+resource (the specific container) on the host.
+"""
+
+import os
+import sys
+
+from osbuild import inputs
+from osbuild.util import containers
+from osbuild.util.mnt import MountGuard
+
+SCHEMA = r"""
+"definitions": {
+  "source-options": {
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "name": {
+        "type": "string",
+        "description": "The name to use for the image"
+      }
+    }
+  },
+  "source-object-ref": {
+    "type": "object",
+    "additionalProperties": false,
+    "minProperties": 1,
+    "patternProperties": {
+      ".*": {
+        "$ref": "#/definitions/source-options"
+      }
+    }
+  },
+  "source-array-ref": {
+    "type": "array",
+    "minItems": 1,
+    "items": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["id"],
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "options": {
+          "$ref": "#/definitions/source-options"
+        }
+      }
+    }
+  },
+  "source-origin": {
+    "type": "string",
+    "description": "When the origin of the input is a source",
+    "enum": ["org.osbuild.source"]
+  }
+},
+"additionalProperties": false,
+"required": ["type", "origin", "references"],
+"properties": {
+  "type": {
+    "enum": ["org.osbuild.containers-storage"]
+  },
+  "origin": {
+    "description": "The org.osbuild.source origin case",
+    "$ref": "#/definitions/source-origin"
+  },
+  "references": {
+    "description": "Container image id",
+     "oneOf": [
+      {"$ref": "#/definitions/source-array-ref"},
+      {"$ref": "#/definitions/source-object-ref"}
+    ]
+  }
+}
+"""
+
+
+class ContainersInput(inputs.InputService):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.mg = MountGuard()
+        self.storage_conf = None
+
+    def map_storage_ref(self, _source, ref, target):
+        # we only want to read the host once and only for inputs that
+        # require it. If we do this for all inputs, reading the host
+        # may cause unexpected errors.
+        if self.storage_conf is None:
+            self.storage_conf = containers.get_host_storage()
+
+        source = self.storage_conf["storage"]["graphroot"]
+        dest = os.path.join(target, "storage")
+
+        # bind mount the input directory to the destination
+        os.makedirs(dest, exist_ok=True)
+        self.mg.mount(source, dest)
+
+        return ref, "containers-storage"
+
+    def map(self, store, origin, refs, target, _options):
+        images = {}
+
+        for ref, data in refs.items():
+            source = "org.osbuild.containers-storage"
+            ref, container_format = self.map_storage_ref(source, ref, target)
```

Comment:
The only `container_format` we return here is `containers-storage` AFAICT so if this is the only one, do we actually need to return this string at all (and later check for it)? Same for ref which is not used. 

This (and the above comment about when we need to load storage.conf) make me wonder if we can just do:
```python
diff --git a/inputs/org.osbuild.containers-storage b/inputs/org.osbuild.containers-storage
index 60fd0621..49b60328 100755
--- a/inputs/org.osbuild.containers-storage
+++ b/inputs/org.osbuild.containers-storage
@@ -84,20 +84,14 @@ SCHEMA = r"""
 """
 
 
-class ContainersInput(inputs.InputService):
+class ContainersStorageInput(inputs.InputService):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.mg = MountGuard()
-        self.storage_conf = None
-
-    def map_storage_ref(self, _source, ref, target):
-        # we only want to read the host once and only for inputs that
-        # require it. If we do this for all inputs, reading the host
-        # may cause unexpected errors.
-        if self.storage_conf is None:
-            self.storage_conf = containers.get_host_storage()
+        self.storage_conf = containers.get_host_storage()
 
+    def bind_mount_local_storage(self, target):
         source = self.storage_conf["storage"]["graphroot"]
         dest = os.path.join(target, "storage")
 
@@ -105,23 +99,17 @@ class ContainersInput(inputs.InputService):
         os.makedirs(dest, exist_ok=True)
         self.mg.mount(source, dest)
 
-        return ref, "containers-storage"
-
     def map(self, store, origin, refs, target, _options):
-        images = {}
+        self.bind_mount_local_storage(target)
 
+        images = {}
         for ref, data in refs.items():
-            source = "org.osbuild.containers-storage"
-            ref, container_format = self.map_storage_ref(source, ref, target)
-
             images[ref] = {
-                "format": container_format,
+                "format": "containers-storage",
                 "name": data["name"]
             }
             images[ref]["name"] = data["name"]
-
-            if container_format == "containers-storage" and self.storage_conf is not None:
-                images[ref]["storage"] = self.storage_conf["storage"]
+            images[ref]["storage"] = self.storage_conf["storage"]
 
         reply = {
             "path": target,
@@ -137,7 +125,7 @@ class ContainersInput(inputs.InputService):
 
 
 def main():
-    service = ContainersInput.from_args(sys.argv[1:])
+    service = ContainersStorageInput.from_args(sys.argv[1:])
     service.main()
 
 

```